src/test/java/core/basesyntax/strategy/handler/impl/PurchaseOperationHandlerTest.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.strategy.handler.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.strategy.handler.OperationHandler;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class PurchaseOperationHandlerTest {
+    private OperationHandler handler;
+    private int oldValue;
+
+    @BeforeEach
+    void setUp() {
+        handler = new PurchaseOperationHandler();
+        oldValue = 500;
+    }
+
+    @Test
+    void operate_validTransaction_ok() {
+        int transaction = 200;
+        int result = handler.operate(transaction, oldValue);
+        assertEquals(oldValue - transaction, result);
```

Comment:
In my opinion, better clearly initialize a variable expected = 300;
```suggestion
          int expected = 300;
          int actual = handler.operate(transaction, oldValue);
        assertEquals(expected, actual);
```